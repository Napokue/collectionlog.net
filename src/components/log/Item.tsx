import { Item as ElementItem } from '../elements';

interface EntryItemProps {
	item: CollectionLogItem;
}

const Item = (props: EntryItemProps) => {
  const { item } = props;

  return (
    <ElementItem item={item} showQuantity={true} showTooltip={true} />
  );
};

export default Item;
