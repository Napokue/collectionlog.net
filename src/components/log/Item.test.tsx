import Item from './Item';
import renderer from 'react-test-renderer';
import { TestUtilities } from '../../tests';

describe(`${Item.name} component`, () => {
  test(
    'should render',
    () => {
      const render = renderer.create(<Item item={TestUtilities.createCollectionLogItem()}/>);
      expect(render).toMatchSnapshot();
    });
});
