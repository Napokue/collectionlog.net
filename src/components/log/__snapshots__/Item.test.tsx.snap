// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should render 1`] = `
<div
  className="w-1/4 sm:w-1/5 md:w-[16.6%] h-[50px] mb-[10px]"
>
  <span
    className="relative top-[25%] left-[2%] md:top-[20%] md:left-[15%] text-yellow"
  >
    1
  </span>
  <div
    className="flex justify-center h-[45px] group"
  >
    <a
      className="hidden flex-col items-center justify-center mb-6 h-full group-hover:flex no-underline"
      href="https://oldschool.runescape.wiki/w/Collection_log"
      rel="noreferrer"
      target="_blank"
    >
      <p
        className="whitespace-nowrap"
      >
        Collection log
      </p>
      <p>
        1/1/2023
      </p>
    </a>
    <img
      className="group-hover:hidden"
      src="data:image/jpeg;charset=utf-8;base64,undefined"
    />
  </div>
</div>
`;
